### ISSUE TITLE
Unnecessary scanning of Text nodes for markers.

### ISSUE THREAD

**Issue by justinfagnani** [Polymer/lit-html#755]
The `Template` constructor scans every Text node for the presence of marker text. In practice, only `<style>` and `<script>` tags should see marker text, as opposed to a marker comment node having been created.

```ts
        } else if (node.nodeType === 3 /* Node.TEXT_NODE */) {
          const data = (node as Text).data!;
          if (data.indexOf(marker) >= 0) {
```

By changing the condition to:

```ts
        } else if (node.nodeType === 3 /* Node.TEXT_NODE */ && 
            node.parentElement && node.parentElement!.tagName === 'STYLE') {
```

all tests still pass.

I think we're probably missing some tests to cover `<script>` tags and possibly some cases where an unescaped `<` could occur in the HTML text, which could make the `lastAttributeNameRegex` fail. If we can get additional tests to pass we should make this optimization.

**Comment by lazystring** [comment]
Mind if I take a stab at this one?

**Comment by justinfagnani** [comment]
This is implemented in 2.0.
